### PR TITLE
Build curl with minimal external libraries.

### PR DESCRIPTION
**Pull request by ViralBShah**

### CODE REVIEW

File: deps/curl.mk [hunk 1]
```diff
@@ -18,7 +18,7 @@ $(SRCDIR)/srccache/curl-$(CURL_VER)/configure: $(SRCDIR)/srccache/curl-$(CURL_VE
 $(BUILDDIR)/curl-$(CURL_VER)/config.status: $(SRCDIR)/srccache/curl-$(CURL_VER)/configure
 	mkdir -p $(dir $@)
 	cd $(dir $@) && \
-	$< $(CONFIGURE_COMMON) --includedir=$(build_includedir) --without-ssl --with-mbedtls=$(build_prefix) CFLAGS="$(CFLAGS) $(CURL_CFLAGS)" LDFLAGS="$(LDFLAGS) $(CURL_LDFLAGS)"
+	$< $(CONFIGURE_COMMON) --includedir=$(build_includedir) --disable-silent-rules --without-gnutls --without-gssapi --without-libidn --without-libmetalink --without-librtmp --without-libssh2 --without-nghttp2 --without-nss --without-polarssl --without-spnego --without-ssl --disable-ares --disable-ldap --disable-ldaps --without-zsh-functions-dir --with-mbedtls=$(build_prefix) CFLAGS="$(CFLAGS) $(CURL_CFLAGS)" LDFLAGS="$(LDFLAGS) $(CURL_LDFLAGS)"
```
Comment (tkelman) [inline deps/curl.mk]: this is getting far too wide for a single line, should be wrapped

Comment (tkelman) [inline deps/curl.mk]: we should probably leave libssh but specify that we want it to link against ours

File: deps/curl.mk [hunk 2]
```diff
@@ -18,7 +18,14 @@ $(SRCDIR)/srccache/curl-$(CURL_VER)/configure: $(SRCDIR)/srccache/curl-$(CURL_VE
 $(BUILDDIR)/curl-$(CURL_VER)/config.status: $(SRCDIR)/srccache/curl-$(CURL_VER)/configure
 	mkdir -p $(dir $@)
 	cd $(dir $@) && \
-	$< $(CONFIGURE_COMMON) --includedir=$(build_includedir) --without-ssl --with-mbedtls=$(build_prefix) CFLAGS="$(CFLAGS) $(CURL_CFLAGS)" LDFLAGS="$(LDFLAGS) $(CURL_LDFLAGS)"
+	$< $(CONFIGURE_COMMON) --includedir=$(build_includedir) \
+	--disable-silent-rules --without-gnutls --without-gssapi \
```
Comment (vtjnash) [inline deps/curl.mk]: we don't want `disable-silent-rules` (it just makes `make` noisy, by defaulting to V=1 instead of V=0)

**Comment by tkelman** [comment]
Please test `make -C deps distclean-foo && make` when you change any of the deps makefiles. A from-scratch build of master fails right now.

**Pull request merged**